snap/validate.go

```diff
@@ -614,6 +614,31 @@ func validateAppTimer(app *AppInfo) error {
 	return nil
 }
 
+func validateAppRestart(app *AppInfo) error {
+	// app.RestartCond value is validated when unmarshalling
+
+	if app.RestartDelay == 0 && app.RestartCond == "" {
+		return nil
+	}
+
+	if app.RestartDelay != 0 {
+		if !app.IsService() {
+			return fmt.Errorf("cannot define restart-delay for application %q as it's not a service", app.Name)
+		}
+
+		if app.RestartDelay < 0 {
+			return fmt.Errorf("cannot use a negative restart-delay for application %q", app.Name)
+		}
+	}
+
+	if app.RestartCond != "" {
+		if !app.IsService() {
+			return fmt.Errorf("cannot define restart-condition for application %q as it's not a service", app.Name)
```

Comment:
`restart-condition is only applicable to services`

Now with all three of those it would be great if we had a way to scope the error in the context of an application. This would require reworking some of the validation to be done per app and encapsulate the error with the app name. Something like `in application "foo": restart-delay cannot be negative, restart-condition is only applicable to services", etc.

Feel free to ignore all three suggestions though, this may require a rework of the error handling logic to implement.